sale_purchase_drop_shipping_asx/models/stock_rule.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2019 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+
+
+class StockRule(models.Model):
+    _inherit = 'stock.rule'
+
+    @api.multi
+    def _run_buy(self, product_id, product_qty, product_uom, location_id, name, origin, values):
+        cache = {}
+        suppliers = product_id.seller_ids\
+            .filtered(lambda r: (not r.company_id or r.company_id == values['company_id']) and (not r.product_id or r.product_id == product_id))
+        if not suppliers:
+            msg = _('There is no vendor associated to the product %s. Please define a vendor for this product.') % (
+                product_id.display_name,)
+            raise UserError(msg)
+        supplier = self._make_po_select_supplier(values, suppliers)
+        partner = supplier.name
+        # we put `supplier_info` in values for extensibility purposes
+        values['supplier'] = supplier
+
+        domain = self._make_po_get_domain(values, partner)
+        # <<< QTL Added
+        # Add sale_shipping_id to the po domain, merge po when the sale_shipping_id are the same
+        sale_order = self.env['sale.order'].search([('name', '=', origin)])
+        if sale_order:
+            domain += (
+                ('sale_shipping_id', '=', sale_order.partner_shipping_id.id),
+            )
+        # >>> QTL Added
```

Comment:
Is there a reason you override `_run_buy()` method instead of extending `_make_po_get_domain()`?